site/content/posts/2022-04-06-april-drumbeat.md

```diff
@@ -0,0 +1,104 @@
+Warnings:
+1627 podlogs/kube-system/kube-controller-manager-kind-control-plane/logs/kube-controller-manager.txt
+ 330 podlogs/kube-system/kube-apiserver-kind-control-plane/logs/kube-apiserver.txt
+  35 podlogs/kube-system/kube-scheduler-kind-control-plane/logs/kube-scheduler.txt
+…
+
+```
+
+> Note: All of this data comes from the automatic queries that Sonobuoy does in addition to your configured plugins and not from the plugins themselves. So errors/warnings are only detected in the configured namespaces (defaults to kube-system and the namespace Sonobuoy is running in).
+
+There are clearly lots of errors and warnings in the control-plane logs, but if you are trying to trace down errors it can be helpful to see if there are an increased number in a particular file. 
+
+The error/warning detection checks for basic string patterns and may be extended or configurable in the future.
+
+## Ongoing Plugin Efforts
+
+As part of [strategy][strategy] we are continuing to develop additional plugins that can be utilized by the community. We wanted to mention two that are currently in development which we hope will find broad support.
+
+### Sonolark
+
+[Sonolark][sonolark] is a plugin that allows you script your own business logic without any of the image management that normally goes with maintaining a plugin. It uses the [Starlark][starlark] language as the scripting language which allows you to take advantage of our base images’ library of features and assertions. It provides access to the Kubernetes API, integration with Sonobuoy, and an expanding library of assertion/translation functions.
+
+This makes a really lightweight solution which can help in numerous situations where you might otherwise simply provide end-users with instructions to follow. In addition, since there is no image management involved, development can be extremely fast since you just tweak the script and have your new plugin ready to use or share.
+
+Sonolark has just had its initial commit so we hope that you will give it a try and help us expand its functionality and make it even better.
+
+### Post-Processor
+
+The [post-processor][postprocessor] is unconventional in that it isn’t a plugin on its own. Instead, it is intended to be used as a sidecar container for any of your other plugins.
+
+The idea is that internally Sonobuoy uses a YAML format to report results and there are occasions where you may want to make modifications to the data after the fact. This is particularly true when you can’t modify the plugin code directly (e.g. for the upstream Kubernetes E2E plugin).
+
+By utilizing [ytt][ytt], the user can provide a transform via a config-map and do things like:
+
+Add additional context to errors such as KB articles
+Remove unnecessary tests to make output more readable (e.g. remove those 5,000 skipped tests from the e2e output)
+Automatically tagging or removing tests known to be flaky while they get fixed
```

Comment:
tag or remove tests (the first two verbs in the list are without ing, so might as well make this parallel)